src/X.PagedList.EF/PagedListExtensions.cs

```diff
@@ -29,7 +29,7 @@ public static class PagedListExtensions
     /// about the collection of objects the subset was created from.
     /// </returns>
     /// <seealso cref="PagedList{T}"/>
-    public static async Task<IPagedList<T>> ToPagedListAsync<T>(this IQueryable<T>? superset, int pageNumber, int pageSize, int? totalSetCount, CancellationToken cancellationToken)
```

Comment:
This project use Nullable settings. So make sense to save '?' option.

Reply:
The point of nullable annotations is to declare intent:
If the variable/argument is supposed to be null, mark it as such (i.e. append the "?"). If it is not supposed to be null, mark it as non-nullable (i.e. omit the "?"). This provides additional information to the user.

For the case at hand:
If we were handling a nullable `superset` gracefully inside the method, it would be okay to have the `superset` argument nullable (e.g. if we returned "null"). However, we throw an Exception, which is equivalent to saying "Do not use null as argument". With nullable annotations, we may explicitly "say" this now by making the argument non-nullable, i.e. by omitting the "?" here.

(If this is used properly throughout the library, the benefit will be that we can detect issues like that at compile time, and do not have to wait for the ArgumentNullException which will happen only at runtime. However, implementing nullable annotations for existing code is quite a lot of work and requires extensive care.)